lib/hsa/mcwamp_hsa.cpp

```diff
@@ -1556,8 +1555,13 @@ class HSAQueue final : public KalmarQueue
 
 
         bool foundFirstValidOp = false;
+        int oldAysncOpsSize = asyncOps.size();
+        int numWaitOps = oldAysncOpsSize - 1;
+        if (drainingQueue_) {
+            numWaitOps = oldAysncOpsSize / 2;
```

Comment:
Please notice this is just a heuristic and might have to be tuned per application. So maybe it's better to expose it as an environment variable.

Reply:
@whchung I have added the environment variable, namely HCC_QUEUE_FLUSHING_RATIO, to tune the percentage of the queue to be flushed when the queue is full.  The percentage has to be greater than zero.  Any invalid value will be set to the default value.  The default HCC_QUEUE_FLUSHING_RATIO is 50, which is 50 percent of the queue.